dali/pipeline/operator/common.h

```diff
@@ -137,19 +132,16 @@ void GetGeneralizedArg(span<T> result, const std::string name, int sample_idx, c
 }
 
 template <typename ArgumentType, typename ExtentType>
-std::pair<int, int> GetShapeLikeArgument(std::vector<ExtentType> &out_shape,
-                                         const OpSpec &spec,
+std::pair<int, int> GetShapeLikeArgument(std::vector<ExtentType> &out_shape, const OpSpec &spec,
                                          const std::string &argument_name,
-                                         const ArgumentWorkspace &ws,
-                                         int ndim = -1,
-                                         int batch_size = -1 /* -1 = get from "batch_size" arg */) {
-  if (batch_size < 0)
-    batch_size = spec.GetArgument<int>("batch_size");
+                                         const ArgumentWorkspace &ws, int ndim = -1,
+                                         int batch_size = -1) {
+  DALI_ENFORCE(batch_size > 0, "Negative (default) batch_size is not supported anymore");
```

Comment:
Likewise - it's not a run-time condition, it's a _coding error_:
```suggestion
  assert(batch_size >= 0);
```